src/initdt.jl

```diff
@@ -3,83 +3,70 @@
   f₀ = zeros(u0./t); f₁ = zeros(u0./t); u₁ = zeros(u0); sk = zeros(u0);
   # Hack to  make a generic u0 with no units, https://github.com/JuliaLang/julia/issues/22216
   typeof(u0[1]) <: AbstractArray ? tmp = zeros(u0,typeof(ones(u0[1]))) : tmp = zeros(u0,typeof(one(u0[1])))
-  uidx = eachindex(u0)
-  #sk = abstol+abs.(u0).*reltol
-  #d₀ = internalnorm(u0./sk)
 
-  @tight_loop_macros for (i,atol,rtol) in zip(uidx,Iterators.cycle(abstol),Iterators.cycle(reltol))
-    @inbounds sk[i] = (atol+abs(u0[i])*rtol)
-    @inbounds tmp[i] = u0[i]./sk[i]
-  end
+  @. sk = abstol+abs(u0)*reltol
+  @. tmp = u0/sk
   d₀ = internalnorm(tmp)
 
   f(t,u0,f₀)
   if any((isnan(x) for x in f₀))
     warn("First function call produced NaNs. Exiting.")
   end
-  #d₁ = internalnorm((f₀./sk*tType(1))/tType(1))
 
-  @tight_loop_macros for i in uidx
-    tmp[i] = (f₀[i]./sk[i]*tType(1))
-  end
+  @. tmp = f₀/sk*oneunit(tType)
   d₁ = internalnorm(tmp)
 
-  T0 = typeof(d₀)
-  T1 = typeof(d₁)
-  if d₀ < T0(1//10^(5)) || d₁ < T1(1//10^(5))
+  if d₀ < 1//10^(5) || d₁ < 1//10^(5)
```

Comment:
Yes, I did run all long tests as well, including units_tests.jl. They all passed.

`tmp` is unitless, and hence both `d₀` and `d₁` are unitless as well.